04-vue-cli/02-TheToaster/components/TheToaster.vue

```diff
@@ -1,24 +1,52 @@
 <template>
   <div class="toasts">
-    <div class="toast toast_success">
-      <UiIcon class="toast__icon" icon="check-circle" />
-      <span>Success Toast Example</span>
-    </div>
-
-    <div class="toast toast_error">
-      <UiIcon class="toast__icon" icon="alert-circle" />
-      <span>Error Toast Example</span>
-    </div>
+    <UiToast v-for="toast in toasts" :key="toast.id" :message="toast.message" :class="`toast_${toast.type}`" :toastType="toast.type" />
```

Comment:
There is still a problem with styles. Classes `toast_success` and `toast_error` are used here but defined in the child component `UiToast`. It is a problem because:
1. It works implicitly, using styles leak in `<style scoped>`
2. It is a non-obvious interface of a component. class is used as a prop, allowing to pass one of 2 supported classes. Component's user must know that exactly these 2 classes can be passed.

Alternatives:
1. Keep these 2-classes styles in the parent. It solves both problems, but makes `toastType` affect only the icon and nothing else. Like there are no different styles of different types of toasts
2. Set classes inside `UiToast` according to its `toastType`.